app.go

```diff
@@ -823,19 +840,28 @@ func (a *Application) deliverTx2(storeTx store.KVStoreTx, txBytes []byte) abci.R
 
 // Commit commits the current block
 func (a *Application) Commit() abci.ResponseCommit {
+
 	var err error
 	defer func(begin time.Time) {
 		lvs := []string{"method", "Commit", "error", fmt.Sprint(err != nil)}
 		committedBlockCount.With(lvs...).Add(1)
 		commitBlockLatency.With(lvs...).Observe(time.Since(begin).Seconds())
 	}(time.Now())
+
+	// Commit EVM state
+	evmStateRoot, err := a.EVMState.Commit(true)
+	if err != nil {
+		panic(err)
+	}
+	a.EvmStore.SetVMRootKey(evmStateRoot[:])
+	a.EVMState.Reset(evmStateRoot)
```

Comment:
Why is this stuff here instead of `EvmStore.Commit`?

Reply:
`EvmStore` is at lower level. `EVMState` is a layer on top of `EVMStore` just like `State` is on top of  AppStore.